lib/main_common.pm

```diff
@@ -2401,116 +2401,186 @@ sub load_virtualization_tests {
-    return unless get_var('VIRT_PART');
-    my $windows = check_var('VIRT_PART', 'windows');
 
-    # Install hypervisor via autoyast or manually
-    loadtest "autoyast/prepare_profile" if get_var "AUTOYAST_PREPARE_PROFILE";
-    load_boot_tests if check_var('VIRT_PART', 'install');
+    # Only proceed if the VIRT_TEST variable is set
+    return unless (get_var('VIRT_TEST'));
 
-    if (get_var("AUTOYAST")) {
-        loadtest "autoyast/installation";
-        loadtest "virt_autotest/reboot_and_wait_up_normal";
-    }
-    else {
-        load_inst_tests if check_var('VIRT_PART', 'install');
+    # Install hypervisor via autoyast or manually
+    unless (get_var('DISABLE_INSTALL')) {
+        if (check_var('HOST_INSTALL_AUTOYAST', 1)) {
+            # Install hypervisor via autoyast
+            loadtest "autoyast/prepare_profile";
+            loadtest "boot/boot_from_pxe";
+            loadtest "autoyast/installation";
+        } else {
+            # Install hypervisor manually
+            loadtest "boot/boot_from_pxe";
+            loadtest "installation/welcome";
+            loadtest "installation/scc_registration";
+            loadtest "installation/addon_products_sle";
+            loadtest "installation/system_role";
+            loadtest "installation/partitioning";
+            loadtest "installation/partitioning_firstdisk";
+            loadtest "installation/partitioning_finish";
+            loadtest "installation/installer_timezone";
+            loadtest "installation/user_settings";
+            loadtest "installation/user_settings_root";
+            loadtest "installation/resolve_dependency_issues";
+            loadtest "installation/installation_overview";
+            loadtest "installation/disable_grub_timeout";
+            loadtest "installation/start_install";
+            loadtest "installation/await_install";
+            loadtest "installation/reboot_after_installation";
+        }
     }
 
-    loadtest "virt_autotest/login_console";
-
-    if (check_var('VIRT_PART', 'install')) {
-        loadtest 'virtualization/universal/prepare_guests';    # Prepare libvirt and install guests
-        loadtest 'virtualization/universal/ssh_hypervisor_init';    # Configure SSH for hypervisor
-        loadtest 'virtualization/universal/waitfor_guests';    # Wait for guests to be installed
-
-        loadtest 'virtualization/universal/ssh_guests_init';    # Fetch SSH key from guests and connect
-        loadtest 'virtualization/universal/register_guests';    # Register guests against the SMT server
-        loadtest 'virtualization/universal/upgrade_guests';    # Upgrade all guests
-        loadtest 'virtualization/universal/patch_guests';    # Apply patches to all compatible guests
-        loadtest 'virtualization/universal/patch_and_reboot';    # Apply updates and reboot
-
+    # QAM VM installation
+    unless (get_var('DISABLE_VM_INSTALL')) {
         loadtest "virt_autotest/login_console";
```

Comment:
This line is good to run before `unless (get_var('DISABLE_VM_INSTALL')) {`? Then no matter which feature test is enabled/disabled, it can start well.

Reply:
For DISABLE_XXXX parameter, no such setting on testsuite, job group, worker setting.
Just use it for test engineer to debug.
Engineer have an idea how to deal with which test module need to start or not.
My opinion.

Reply:
I am afraid this is not fully correct. For example, we are going to have a separate testsuite for sriov, which will have DISABLE_XX in testsuite setting to disable all other feature tests, except sriov.

Reply:
No, no, no. There is separate ticket for sriov, see https://progress.opensuse.org/issues/161192. As long as sriov test can run well in the old way, it is fine to  not think about it at all in your current ticket :).

Reply:
I've also added an SR-IOV test. Please review it concurrently.
it is convenient to think about it at the same time.

Reply:
@RoyCai7 Hi Roy, thanks for adding this. But please be aware that if you modify it in this PR, it means that all requirements in the description of https://progress.opensuse.org/issues/161192 have to be handled in the PR and also on void.qam.suse.cz job groups, testsuites, etc, which is actually not necessary. Do you want to do it? If not, suggest to recover it and just ensures that original way works for sriov(I think it should work because nothing is broken with your PR, just add some regression job to prove it). If yes, please think about it overall.

Reply:
so many comment, the message is folded.
@alice-suse  No matter what, even if we don't address the SR-IOV part now, we'll still need to consider integrating SR-IOV tests in the current PR code later. So, it makes sense to consider it now and handle it all together. thanks for your reminder.

Reply:
It's okay this time. Generally I'd recommend to not break the boundary between progress tasks :) .